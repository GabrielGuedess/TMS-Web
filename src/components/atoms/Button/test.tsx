import { render, screen } from '@testing-library/react';

import { Button } from '.';

describe('<Button />', () => {
  it('should render the button correctly', () => {
    const { container } = render(<Button />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the button is loading', () => {
    render(<Button isLoading />);

    const loading = screen.getByLabelText('Loading');

    expect(loading).toBeInTheDocument();
  });

  it('should render the button with children', () => {
    render(<Button>Button</Button>);

    const button = screen.getByText('Button');

    expect(button).toBeInTheDocument();
  });

  it('should render the button with color dark', () => {
    render(
      <Button color="dark" variant="solid">
        Button
      </Button>,
    );

    const button = screen.getByLabelText('Button Background');

    expect(button).toHaveClass('bg-shark-950');
  });

  it('should render the button with color info', () => {
    render(
      <Button color="info" variant="solid">
        Button
      </Button>,
    );

    const button = screen.getByLabelText('Button Background');

    expect(button).toHaveClass('bg-cobalt-700');
  });

  it('should render the button with color danger', () => {
    render(
      <Button color="danger" variant="solid">
        Button
      </Button>,
    );

    const button = screen.getByLabelText('Button Background');

    expect(button).toHaveClass('bg-danger-500');
  });

  it('should render the button with color success', () => {
    render(
      <Button color="success" variant="solid">
        Button
      </Button>,
    );

    const button = screen.getByLabelText('Button Background');

    expect(button).toHaveClass('bg-success-500');
  });

  it('should render the button with color warning', () => {
    render(
      <Button color="warning" variant="solid">
        Button
      </Button>,
    );

    const button = screen.getByLabelText('Button Background');

    expect(button).toHaveClass('bg-warning-500');
  });

  it('should render the button with color secondary', () => {
    render(
      <Button variant="solid" color="secondary">
        Button
      </Button>,
    );

    const button = screen.getByLabelText('Button Background');

    expect(button).toHaveClass('bg-dark-200');
  });
});
