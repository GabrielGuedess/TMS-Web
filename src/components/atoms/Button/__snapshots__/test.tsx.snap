// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the button correctly 1`] = `
<button
  class="relative transition-transform duration-200 ease-in-out hover:scale-105 focus:scale-105 active:scale-90"
  type="button"
>
  <div
    aria-label="Button Background"
    class="absolute size-full rounded-md bg-primary-400"
  />
  <div
    class="flex h-full items-center justify-center gap-2 px-5 py-[0.625rem] flex-row text-white"
  />
</button>
`;
